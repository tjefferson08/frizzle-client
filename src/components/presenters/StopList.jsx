import React from 'react';
import StopCard from './StopCard';

const StopList = ({ stops }) => {
  return (
    <ul>
      {stops.map((stop, i) => <StopCard key={i} {...stop} />)}
    </ul>
  );
};

export default StopList;
