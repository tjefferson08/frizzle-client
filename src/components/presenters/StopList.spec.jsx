import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import StopList from './StopList';

test('should render a UL of stops', () => {
  const renderer = ReactTestUtils.createRenderer();
  const fakeStops = [{ stop_name: 'Stop 1' }, { stop_name: 'Stop 2' }];
  renderer.render(<StopList stops={fakeStops} />);
  const result = renderer.getRenderOutput();

  expect(result.type).toBe('ul');
  expect(result).toMatchSnapshot();
});

