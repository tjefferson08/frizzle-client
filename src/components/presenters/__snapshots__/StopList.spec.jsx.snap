exports[`test should render a UL of stops 1`] = `
<ul>
  <Stop
    stop_name="Stop 1" />
  <Stop
    stop_name="Stop 2" />
</ul>
`;
